packages/reakit/src/Provider/Provider.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from "react";
+import * as PropTypes from "prop-types";
+import {
+  Provider as ConstateProvider,
+  ProviderProps as ConstateProps
+} from "constate";
+import { ThemeProvider, ThemeProviderProps } from "../styled";
+
+interface ComponentProps
+  extends ConstateProps<{ [key: string]: any }>,
+    ThemeProviderProps<Object> {
+  children: React.ReactNode;
+}
+
+const Provider: React.SFC<ComponentProps> = ({ theme, children, ...props }) => (
```

Comment:
I don't know how to put the generic `S` here. Maybe transforming it into a class?